File: mono/src/symbol/property.rs
use serde::{Deserialize, Serialize};

use crate::symbol::Declaration;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Property {
    #[serde(flatten)]
    pub declaration: Declaration,

    /// Type of the property
    pub r#type: String,

    /// Whether getter exists
    pub getter: bool,

    /// Whether setter exists
    pub setter: bool,
}

File: typings/symbols/property.ts
import { Declaration } from './base';
import { Identifier } from './types';

export interface Property extends Declaration {
    tag: Identifier.Property,

    /**
     * @brief Type of the property
     * @readonly
     */
    readonly type: string,

    /**
     * @brief Whether getter exists
     * @readonly
     */
    readonly getter: boolean,

    /**
     * @brief Whether setter exists
     * @readonly
     */
    readonly setter: boolean,
}
